### DISCUSSION TITLE
pycapnp only installs capnpc-cython.exe

### DISCUSSION THREAD

**Discussion by coreyhartley** [capnproto/capnproto#1947] [Q&A] [answered]
I've tried installing several times over, both using pip and the repository.  Both result in the same outcome.  I'm using `python 3.11` on my machine.  I don't have anything related to `cython` on my machine.

When installed, I've tried running:
```
 capnp compile -o python
```
resulting in `python: no such plugin (executable should be 'capnpc-python')`

In desperation, I've also tried:
```
capnp compile -o cython:./python ./schema/ut.capnp
```
resulting in:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\corey.hartley\AppData\Local\Python\Python311\Scripts\capnpc-cython.exe\__main__.py", line 7, in <module>
  File "C:\Users\corey.hartley\AppData\Local\Python\Python311\Lib\site-packages\capnp\_gen.py", line 41, in main
    for field in node["struct"]["fields"]:
                 ~~~~~~~~~~~~~~^^^^^^^^^^
KeyError: 'fields'
cython: plugin failed: exit code 1
```

Is there anyone that could provide some guidance?  What am I doing wrong?

**Comment by haata** [comment, marked as answer]
For pycapnp, .capnp are processed on the fly. I'm not sure the last time
capnp compile -o cython has been tested (it hasn't been at least as long as
I've worked on it).

**Reply by coreyhartley** [reply]
I just started to figure that out after going through the Quickstart portion of the documentation.  I'm going to close the discussion.  Thank you for the reply!
